packages/hash/frontend/src/blocks/page/PageTitle/utils.ts

```diff
@@ -0,0 +1,28 @@
+import { Schema } from "prosemirror-model";
+import { Selection } from "prosemirror-state";
+import { EditorView } from "prosemirror-view";
+
+// TODO: Improve page title validation and use it when creating pages.
+// Alternatively, we can validate on server-side only and handle mutation errors.
+export const isValidPageTitle = (value: string): boolean =>
+  Boolean(value.length);
+
+export const cleanUpTitle = (value: string): string =>
+  value.trim().replace(/\s+/g, " ");
+
+export const focusEditorBeginning = (view?: EditorView<Schema>) => {
+  if (!view) return;
+
+  const { state } = view;
+
+  const newSelection = Selection.atStart(state.doc);
+
+  const tr = state.tr.setSelection(newSelection);
+
+  const newState = state.apply(tr);
+  view.updateState(newState);
+
+  setImmediate(() => {
```

Comment:
What's this setImmediate necessary for?

Reply:
```
* if we don't wait with setImmediate here, new view selection does not work correctly,
* and we land focus on 2nd node instead of 1st on editor
   ```

Added comment line explaining it here https://github.com/hashintel/hash/pull/956/commits/d24cca980c0e4486d8001a4aee90312a72013c2c